include/baresip.h

```diff
@@ -832,8 +833,11 @@ int  uag_reset_transp(bool reg, bool reinvite);
 void uag_set_sub_handler(sip_msg_h *subh);
 int  uag_set_extra_params(const char *eprm);
 struct ua   *uag_find(const struct pl *cuser);
+struct ua   *uag_find_msg(const struct sip_msg *msg);
 struct ua   *uag_find_aor(const char *aor);
 struct ua   *uag_find_param(const char *name, const char *val);
+struct ua   *uag_find_connect(const struct pl *req_uri);
```

Comment:
perhaps rename "connect" to something else, for example "target".

this is a generic function, and should be possible to use for any outgoing request,
for example OPTIONS:

```c
static int options_command(struct re_printf *pf, void *arg)
{
	const struct cmd_arg *carg = arg;
	struct ua *ua = carg->data ? carg->data : menu_uacur();
	int err = 0;

	(void)pf;

	err = ua_options_send(ua, carg->prm,
			      options_resp_handler, NULL);
	if (err) {
		warning("menu: ua_options failed: %m\n", err);
	}

	return err;
}
```